### DISCUSSION TITLE
How to implement subcategories?

### DISCUSSION THREAD

**Discussion by RodenLuo** [strapi/strapi#6813] [Legacy Topics]
The only related post I found is this [issue](https://github.com/strapi/strapi/issues/5385). Under "Category", I created a component that has a child, which is a relation with "Category" itself. 

![image](https://user-images.githubusercontent.com/8460424/85775446-7522a100-b728-11ea-8734-d3e3d4e2e5b3.png)

I can then add, for example, "Image Colorization" as a subcategory of "Image Processing". 

![image](https://user-images.githubusercontent.com/8460424/85776306-3fca8300-b729-11ea-9bf4-833c08edc463.png)

But when I add a new content record that has a relation to the category, I do not see the hierarchy. Everything is flat. 

![image](https://user-images.githubusercontent.com/8460424/85776660-96d05800-b729-11ea-9313-1c3bf213c268.png)

I hope I can see something like below when I choose the category.

First_Level_Category_1
--Second_Level_category_1
--Second_Level_category_2
First_Level_Category_2
--Second_Level_category_3
--Second_Level_category_4

Thanks,

**Comment by lauriejim** [comment]
Hello! If you want to add this in your app, you will have to customize the Content Manager plugin (the admin part)
Here is the documentation - https://strapi.io/documentation/v3.x/concepts/customization.html#plugin-extensions
Here is the source code - https://github.com/strapi/strapi/tree/master/packages/strapi-plugin-content-manager/admin/src
Find the compo that display the relation and add the logic / design to match your needs
